modules/ml/src/test/java/org/apache/ignite/ml/nn/MLPTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.nn;
+
+import org.apache.ignite.ml.TestUtils;
+import org.apache.ignite.ml.math.Matrix;
+import org.apache.ignite.ml.math.Vector;
+import org.apache.ignite.ml.math.impls.matrix.DenseLocalOnHeapMatrix;
+import org.apache.ignite.ml.math.impls.vector.DenseLocalOnHeapVector;
+import org.apache.ignite.ml.nn.architecture.MLPArchitecture;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for Multilayer perceptron.
+ */
+public class MLPTest {
+    /**
+     * Tests that MLP with 2 layer, 1 neuron in each layer and weight equal to 1 is equivalent to sigmoid function.
+     */
+    @Test
+    public void testSimpleMLPPrediction() {
+        MLPArchitecture conf = new MLPArchitecture(1).withAddedLayer(1, false, Activators.SIGMOID);
+
+        MLP mlp = new MLP(conf, new MLPConstInitializer(1));
+
+        int input = 2;
+
+        Matrix predict = mlp.apply(new DenseLocalOnHeapMatrix(new double[][] {{input}}));
+
+        Assert.assertEquals(predict, new DenseLocalOnHeapMatrix(new double[][] {{Activators.SIGMOID.apply(input)}}));
+    }
+
+    /**
+     * Test that MLP with parameters that should produce function close to 'XOR' is close to 'XOR'.
```

Comment:
What's written there?

Reply:
There are two functions: XOR and MLP with some given parameters. Both take 2d point as input and output 1d point. We can measure how close are they on given domain (Elaborated that we use specific domain in comment).